syntax = "proto3";

package mes_grpc;

import "google/protobuf/timestamp.proto";

service MessengerServer {
    rpc SendMessage(SendRequest) returns (SendResponse);
    rpc ReadMessages(ReadRequest) returns (stream MessageFromStreamRead);
}

message SendRequest {
    string author = 1;
    string text = 2;
}

message SendResponse {
    google.protobuf.Timestamp sendTime = 1;
}

message ReadRequest {}

message MessageFromStreamRead {
    string author = 1;
    string text = 2;
    google.protobuf.Timestamp sendTime = 3;
}
